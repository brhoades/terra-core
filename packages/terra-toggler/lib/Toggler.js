'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

require('terra-base/lib/baseStyles');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  /**
   * Content in the body of the panel that will be expanded or collapsed
   **/
  children: _propTypes2.default.node.isRequired,
  /**
   * Expands or collapses content
   **/
  isOpen: _propTypes2.default.bool
};

var defaultProps = {
  children: null,
  isOpen: false
};

var Toggler = function Toggler(_ref) {
  var isOpen = _ref.isOpen,
      children = _ref.children,
      customProps = _objectWithoutProperties(_ref, ['isOpen', 'children']);

  var togglerClass = (0, _classnames2.default)(['terra-Toggler', customProps.className]);

  if (!isOpen) {
    return null;
  }

  return _react2.default.createElement(
    'div',
    _extends({}, customProps, { className: togglerClass }),
    children
  );
};

Toggler.propTypes = propTypes;
Toggler.defaultProps = defaultProps;

exports.default = Toggler;