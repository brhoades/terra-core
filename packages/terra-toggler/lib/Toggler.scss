@import '~terra-mixins';
@import './variables';
@import './mixins';

// The accordion content panel doesn't have the header highlight, but the other ones do
.terra-Toggler-header {
  @include terra-text-align-start();
  background-color: transparent;
  border: 0;
  cursor: pointer;
  line-height: 20px;
  padding: 0;
  width: 100%;
}

.terra-Toggler-content {
  overflow: hidden;
}

.terra-Toggler.is-collapsed {
  > .terra-Toggler-content {
    max-height: 0;
    transition: max-height 0.25s ease-out;
  }
}

.terra-Toggler.is-expanded {
  > .terra-Toggler-content {
    max-height: 1000px;
    transition: max-height 0.25s ease-in;
  }
}
