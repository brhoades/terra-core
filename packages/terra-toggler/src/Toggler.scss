@import '~terra-mixins';
@import './variables';
@import './mixins';

// The accordion content panel doesn't have the header highlight, but the other ones do
.terra-Toggler-header {
  background-color: transparent;
  border: 0;
  cursor: pointer;
  padding: 0;
  text-align: inherit;
  width: 100%;
}

.terra-Toggler-content {
  overflow: hidden;
}

.terra-Toggler--is-collapsed {
  > .terra-Toggler-content {
    max-height: 0;
    transition: max-height 0.25s ease-out;
  }
}

.terra-Toggler--is-open {
  > .terra-Toggler-content {
    max-height: 1000px;
    transition: max-height 0.25s ease-in;
  }
}
