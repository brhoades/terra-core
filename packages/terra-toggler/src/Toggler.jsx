import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import 'terra-base/lib/baseStyles';

const propTypes = {
  /**
   * Content in the body of the panel that will be expanded or collapsed
   **/
  children: PropTypes.node.isRequired,
  /**
   * Expands or collapses content
   **/
  isOpen: PropTypes.bool,
};

const defaultProps = {
  children: null,
  isOpen: false,
};

const Toggler = ({ isOpen, children, ...customProps }) => {
  const togglerClass = classNames([
    'terra-Toggler',
    customProps.className,
  ]);

  if (!isOpen) {
    return null;
  }

  return (
    <div {...customProps} className={togglerClass} >
      {children}
    </div>
  );
};

Toggler.propTypes = propTypes;
Toggler.defaultProps = defaultProps;

export default Toggler;
