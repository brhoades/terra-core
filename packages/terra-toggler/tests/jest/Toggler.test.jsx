import React from 'react';
import Toggler from '../../src/Toggler';

// Snapshot Tests
it('should render a default toggler', () => {
  const toggler = shallow(<Toggler />);
  expect(toggler).toMatchSnapshot();
});
