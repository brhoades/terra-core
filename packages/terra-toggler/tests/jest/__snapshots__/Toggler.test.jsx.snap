exports[`Controlled Toggler should mount a default toggler 1`] = `
<Toggler
  isOpen={false} />
`;

exports[`Controlled Toggler should render a default toggler 1`] = `null`;

exports[`Controlled Toggler should shallow a default toggler 1`] = `null`;

exports[`Uncontrolled Toggler should mount a default toggler 1`] = `
<TogglerUncontrolled
  buttonContent={null}
  isOpen={false}>
  <div>
    <Button
      aria-expanded={false}
      className="terra-Toggler-button"
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isReversed={false}
      onClick={[Function]}
      variant="default">
      <button
        aria-disabled={false}
        aria-expanded={false}
        className="terra-Button terra-Button--default terra-Toggler-button"
        disabled={false}
        onClick={[Function]} />
    </Button>
    <Toggler
      className="terra-Toggler-uncontrolled"
      isOpen={false} />
  </div>
</TogglerUncontrolled>
`;

exports[`Uncontrolled Toggler should render a default toggler 1`] = `
<div>
  <button
    aria-disabled="false"
    aria-expanded="false"
    class="terra-Button terra-Button--default terra-Toggler-button" />
</div>
`;

exports[`Uncontrolled Toggler should shallow a default toggler 1`] = `
<div>
  <Button
    aria-expanded={false}
    className="terra-Toggler-button"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isReversed={false}
    onClick={[Function]}
    variant="default" />
  <Toggler
    className="terra-Toggler-uncontrolled"
    isOpen={false}>
    <div>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur malesuada dui sed porttitor interdum. Sed scelerisque tempor massa, id mollis enim dapibus sit amet. Nunc interdum orci at venenatis condimentum. Donec scelerisque leo eget aliquam tempus. Nunc maximus rhoncus sem, pharetra gravida risus consectetur et. Ut nec mi eros. Cras convallis pellentesque neque ac tristique. Curabitur iaculis erat ac nulla accumsan mattis.
      </p>
       
      <p>
        In ornare fringilla mauris, non vestibulum orci sodales et. Morbi finibus efficitur vulputate. Cras viverra, felis sit amet porttitor tristique, turpis lacus sodales risus, id fermentum nisl lacus non mauris. Phasellus eu dolor justo. Quisque tempor est purus, et scelerisque justo vestibulum vel. Fusce ornare venenatis nibh, sed tristique ipsum luctus vel. Pellentesque in bibendum lorem. Integer mi metus, aliquam eget turpis quis, consectetur placerat diam. Etiam velit felis, viverra eu ultricies sed, feugiat id nisl. Aliquam vestibulum tincidunt felis vitae scelerisque. Nullam rutrum justo ac maximus sodales.
      </p>
      <p>
        Interdum et malesuada fames ac ante ipsum primis in faucibus. Vestibulum vel arcu suscipit, aliquet lectus scelerisque, tincidunt nunc. Aliquam ipsum nibh, egestas nec consectetur sit amet, suscipit eget dolor. Duis vitae turpis tempor, elementum mauris dapibus, elementum enim. Duis sodales enim volutpat leo ultrices, sed lacinia mauris sodales. Curabitur vitae viverra dolor. Pellentesque tempus, metus eget tristique malesuada, lectus tortor pulvinar nunc, at hendrerit sem odio placerat ex. Nullam efficitur urna eget sem aliquet hendrerit. Donec imperdiet ante erat, vitae condimentum arcu rhoncus sed.
      </p>
    </div>
  </Toggler>
</div>
`;
